libbeat/publisher/queue/memqueue/broker.go

```diff
@@ -135,14 +135,15 @@ func NewQueue(
 	logger *logp.Logger,
 	ackCallback func(eventCount int),
 	settings Settings,
+	inputQueueSize int,
```

Comment:
It's a bit odd not to have `inputQueueSize` be part of `settings` as before. I agree with removing the special override in the publisher pipeline code but I think that can be done while leaving `inputQueueSize` as part of `settings` so this constructor here has still has the same signature as before, with all queue settings being in `settings`.  For that you'll need to set `settings.InputQueueSize` wherever you call this constructor from (unless, of course, you want to leave that field at it's zero-value, e.g. in test code).

Reply:
The only thing I don't like about that is it is hard to tell if you are supposed to set `settings.InputQueueSize` before you pass it into `FactoryForSettings` or allow `FactoryForSettings` to set it.  Also what happens if you pass in a Settings with that field set?  Do you override or keep?  

I don't like changing the signature, but I'm not fond of the ambiguous nature of who sets InputQueueSize if we leave it in settings.  I can be convinced to go either way, do you have a strong opinion?